lib/SILOptimizer/Mandatory/TFConstExpr.cpp

```diff
@@ -1298,6 +1316,19 @@ ConstExprFunctionState::evaluateFlowSensitive(SILInstruction *inst) {
     return computeFSStore(stored, inst->getOperand(1));
   }
 
+  if (auto *dai = dyn_cast<UncheckedTakeEnumDataAddrInst>(inst)) {
```

Comment:
Is this code needed?  In its absence, control will flow down through to call getAddressValue() and will call the hunk above.  It seems that that would be sufficient to handle this in the flow sensitive case.

Reply:
Forgot to update: you are right! Removed this chunk of code. Thanks.